File: leetcode/C++/merge-intervals.cpp
class Solution {
public:
    vector<vector<int>> merge(vector<vector<int>>& intervals) {
        vector<vector<int>> sol;
        sort(intervals.begin(),intervals.end());
        for(int i=0; i<intervals.size(); ++i){
            if(sol.empty() || sol.back()[1]<intervals[i][0]) sol.push_back(intervals[i]);
            else sol.back()[1]=max(sol.back()[1],intervals[i][1]);
        }
        return sol;
    }
};

/*
class Solution {
public:
    vector<vector<int>> merge(vector<vector<int>>& intervals) {
        vector<vector<int>> sol;
        sort(intervals.begin(),intervals.end());
        for(vector<int> &v : intervals){
            if(sol.empty() || sol.back()[1] < v[0]) sol.push_back(v);
            else if(sol.back()[1]<v[1]) sol.back()[1]=v[1];
        }
        return sol;
    }
};
*/

File: leetcode/Python/merge-intervals.py
class Solution:
    def merge(self, intervals: List[List[int]]) -> List[List[int]]:
        sol = []
        intervals.sort()
        for i in intervals:
            if not sol or sol[-1][1]<i[0]:
                sol.append(i)
            else:
                sol[-1][1]=max(sol[-1][1],i[1])
        return sol

'''
class Solution:
    def merge(self, intervals: List[List[int]]) -> List[List[int]]:
        sol = list()
        intervals.sort()
        for interval in intervals:
            if len(sol) == 0 or sol[-1][1] < interval[0]:
                sol.append(interval)
            elif sol[-1][1] < interval[1]:
                sol[-1][1] = interval[1]
        return sol
        
'''